scripts/sct_flatten_sagittal.py

```diff
@@ -64,7 +64,8 @@ def main(fname_anat, fname_centerline, degree_poly, centerline_fitting, interp,
 
     # loop across slices and apply translation
     im_anat_flattened = im_anat.copy()
-    im_anat_flattened.changeType('uint16')  # force uint16 because outputs are converted using img_as_uint()
+    # change type to float32 because of subsequent conversion (img_as_float). See #1790
+    im_anat_flattened.changeType('float32')
```

Comment:
```
try: 
    im_anat_flattened.changeType('uint16')
except ValueError:
    im_anat_flattened.changeType('float32')

```

Reply:
that would not work because `im_anat_flattened.changeType('uint16')` does not produce an error.

Reply:
Do you mean that the subsequent conversion is always *img_as_float* ?

Reply:
yes, we always do img_as_float, therefore the dtype need to be prepared, otherwise if an [0,1] float array is input, we end up with a matrix full of zero.